internal/expr/prepare.go

```diff
@@ -88,15 +88,24 @@ func prepareInput(ti typeNameToInfo, p *inputPart) (tm typeMember, err error) {
 		}
 	}()
 
-	info, ok := ti[p.sourceType.typeName]
+	info, ok := ti[p.sourceType.TypeName()]
 	if !ok {
-		return nil, typeMissingError(p.sourceType.typeName, getKeys(ti))
+		return nil, typeMissingError(p.sourceType.TypeName(), getKeys(ti))
 	}
 
-	tm, err = info.typeMember(p.sourceType.memberName)
-	if err != nil {
-		return nil, err
+	switch t := p.sourceType.(type) {
+	case keyedAccesor:
+		if t.memberName == "*" {
+			return nil, fmt.Errorf(`asterisk used with %s in invalid context`, info.typ().Kind())
```

Comment:
Is this case ever hit? Don't we still have the check for an asterisk in `parseInputExpr`?

Reply:
You are right, removing it here. That is the downside of this type system, I am not sure if we have to check or not without following the logic in the code.

Reply:
Yea, I would like at some point to tighten up the types in parser so that information like this does not have to be re-validated in the prepare stage and is instead stored in parser types.